plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveBucketProperty.java

```diff
@@ -66,11 +67,18 @@ public static Optional<HiveBucketProperty> fromStorageDescriptor(Map<String, Str
         List<SortingColumn> sortedBy = ImmutableList.of();
         if (storageDescriptor.isSetSortCols()) {
             sortedBy = storageDescriptor.getSortCols().stream()
-                    .map(order -> SortingColumn.fromMetastoreApiOrder(order, tablePartitionName))
+                    .map(order -> SortingColumn.fromMetastoreApiOrder(order, tablePartitionName, dataColumns))
                     .collect(toImmutableList());
         }
         BucketingVersion bucketingVersion = HiveBucketing.getBucketingVersion(tableParameters);
-        return Optional.of(new HiveBucketProperty(storageDescriptor.getBucketCols(), bucketingVersion, storageDescriptor.getNumBuckets(), sortedBy));
+        List<String> bucketColumnNames = storageDescriptor.getBucketCols().stream()
+                // Ensure that the names used for the bucket columns are the same as the names used for the table columns
+                .map(bucketColumnName -> dataColumns.stream().filter(column -> column.getName().equalsIgnoreCase(bucketColumnName))
+                        .findFirst()
+                        .orElseThrow(() -> new TrinoException(HIVE_INVALID_METADATA, "Table/partition metadata has invalid bucket column: " + bucketColumnName))
+                        .getName())
+                .toList();
```

Comment:
`.collect(toImmList())` (otherwise `HiveBucketProperty` will copy the list again)